cinema/serializers.py

```diff
@@ -68,13 +84,87 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
+        )
+
+
+class MovieSessionOrderSerializer(serializers.ModelSerializer):
+    movie_title = serializers.SlugRelatedField(
+        many=False,
+        read_only=True,
+        slug_field="title",
+        source="movie"
+    )
+    cinema_hall_name = serializers.SlugRelatedField(
+        many=False,
+        read_only=True,
+        slug_field="name",
+        source="cinema_hall",
+    )
+    cinema_hall_capacity = serializers.SlugRelatedField(
+        many=False,
+        read_only=True,
+        slug_field="capacity",
+        source="cinema_hall"
+    )
+
+    class Meta:
+        model = MovieSession
+        fields = (
+            "id",
+            "show_time",
+            "movie_title",
+            "cinema_hall_name",
+            "cinema_hall_capacity"
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+
+class TicketRowSeatSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketRowSeatSerializer(
+        many=True,
+        read_only=True,
+        source="tickets"
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=False, allow_empty=False)
+
+    class Meta:
+        model = Order
+        fields = ("id", "tickets", "created_at")
+
+    def create(self, validated_data):
+        with transaction.atomic():
```

Comment:
Use transaction.atomic() as decorator instead of context manager, it will make your code more readable